lib/m2ts/caption-stream.js

```diff
@@ -628,14 +671,35 @@ Cea708Stream.prototype.initService = function(serviceNum, i) {
  */
 Cea708Stream.prototype.handleText = function(i, service, isExtended) {
   var packetData = this.current708Packet.data;
-  var b = packetData[i];
   var extended = isExtended ? 0x1000 : 0x0000;
-  var char = get708CharFromCode(extended | b);
+  var currentByte = packetData[i];
+  var previousByte = packetData[i - 1];
+  var nextByte = packetData[i + 1];
   var win = service.currentWindow;
+  var char;
+  var charCodeArray;
+
+  // Use the TextDecoder if one was created for this service
+  if (service.textDecoder_) {
+    // If preceded by a code in the 0x18-0x1f range, this byte and the following byte make up a two-byte character
+    if (previousByte >= 0x18 && previousByte <= 0x1f && nextByte) {
+      charCodeArray = [currentByte, nextByte];
+
+      // Skip the next byte in the loop since it has already been handled
+      i++;
+    } else {
+     charCodeArray = [currentByte];
+    }
+
+    char = service.textDecoder_.decode(new Uint8Array(charCodeArray));
+  } else {
+    char = get708CharFromCode(extended | currentByte);
```

Comment:
so, `get708CharFromCode` method seems to account for some extended characters. We probably need to do the same for the encoded stuff too.
I wonder whether moving the text decoder piece into `flushDisplayed` would work. The benefit is that it could decode once per cue rather than once per character. Unfortunately, didn't get a chance to try it out.

Reply:
I fixed the flow to only use a `TextDecoder` if a `P16` code is encountered (indicating a multibyte character), but to still use `get708CharFromCode()` when an `EXT1` code is encountered (indicating a character from an extended code set). This should address your concern.